x-pack/plugins/snapshot_restore/public/plugin.ts

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { PLUGIN } from '../common/constants';
+import { CLIENT_BASE_PATH, renderReact } from './app';
+import template from './index.html';
+import { createShim } from './shim';
+
+const REACT_ROOT_ID = 'snapshotRestoreReactRoot';
+
+export class Plugin {
+  public start(): void {
+    const shim = createShim();
+    const {
+      core: { i18n, routing, http },
+      plugins: { management },
+    } = shim;
+
+    // Register management section
+    const esSection = management.sections.getSection('elasticsearch');
+    esSection.register(PLUGIN.ID, {
+      visible: true,
+      display: i18n.translate('xpack.snapshotRestore.appName', {
+        defaultMessage: 'Snapshot and Restore',
+      }),
+      order: 7,
+      url: `#${CLIENT_BASE_PATH}/repositories`,
+    });
+
+    // Register react root
+    routing.registerAngularRoute(`${CLIENT_BASE_PATH}/:section?/:subsection?/:view?/:id`, {
```

Comment:
Can you explain why this needs to be an angular route/controller rather than just being react like the rest of the plugin?

Reply:
since the Kibana chrome is using Angular routing, we need to register the initial base path with Angular so it's accessible from the management menu. we then mount the React app inside it which handles the rest of internal stuff in React.

Reply:
So if the management app itself was built using something like react router rather angular router, this wouldn't be necessary here, is that correct?